### PR TITLE
Add support for .NET 6.0

### LINKED ISSUE #1641: OTD does not build nor run with linux distros that upgraded to .NET 6
## Description

OpenTabletDriver does not have a target framework entry to support .NET 6, and the .NET 6 runtime does not have backwards compatibility with .NET 5.

## Possible fixes

- Both frameworks can be supported by specifying both `.net5.0` and `.net6.0` in `TargetFrameworks`
- Just merge #1441 
- Or, embed the runtime and enable assembly trimming for smaller binary file sizes.

### PR DESCRIPTION
**Pull request by X9VoiD**
and use Directory.Build.props to deduplicate some properties.

- Fixes #1641

### CODE REVIEW

File: Directory.Build.props [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<Project>
+  <PropertyGroup>
+    <OTDVersion>0.6.0.0</OTDVersion>
+    <OTDFrameworkBase>net6.0</OTDFrameworkBase>
```
Comment (InfinityGhost) [inline Directory.Build.props:4]: Can we avoid the OTD acronym within the project?

File: OpenTabletDriver.Desktop/Reflection/Metadata/PluginMetadata.cs [hunk 2]
```diff
@@ -68,7 +68,7 @@ public class PluginMetadata
 
         public static string GetSHA256(Stream stream)
         {
-            using (var sha256 = SHA256Managed.Create())
+            using (var sha256 = System.Security.Cryptography.SHA256.Create())
```
Comment (InfinityGhost) [inline OpenTabletDriver.Desktop/Reflection/Metadata/PluginMetadata.cs:71]: Really shouldn't need the full namespace because its included in an using statement
```suggestion
            using (var sha256 = SHA256.Create())
```

Comment (X9VoiD) [reply OpenTabletDriver.Desktop/Reflection/Metadata/PluginMetadata.cs:71]: It conflicts with member "SHA256"

File: Directory.Build.props (hunk 1 above)
Comment (X9VoiD) [reply Directory.Build.props:4]: So... `VersionBase` and `FrameworkBase`?

Comment (InfinityGhost) [reply Directory.Build.props:4]: Works for me

File: OpenTabletDriver.Desktop/Reflection/Metadata/PluginMetadata.cs (hunk 2 above)
Comment (InfinityGhost) [reply OpenTabletDriver.Desktop/Reflection/Metadata/PluginMetadata.cs:71]: Good point, forgot we had our own impl

File: .github/workflows/dotnet.yml [hunk 3]
```diff
@@ -49,16 +49,17 @@ jobs:
       - name: Setup .NET Core
         uses: actions/setup-dotnet@v1
         with:
-          dotnet-version: 5.0.x
+          dotnet-version: '6.0'
+          include-prerelease: True
       - name: MacOS Publish
         env:
           runtime: osx-x64
           framework: net5
```
Comment (AbstractQbit) [inline .github/workflows/dotnet.yml:57]: This is now unused here and in `Windows Publish` too. Might also move the runtime thing into options everywhere while at it

**Comment by gonX** [comment]
Since #1556 was merged, you'll need to adjust `./build.sh` to use net6 as well (once you rebase, anyway)

File: OpenTabletDriver.Plugin/OpenTabletDriver.Plugin.csproj [hunk 4]
```diff
@@ -1,17 +1,13 @@
...
   <PropertyGroup Label="Project Properties">
-    <TargetFrameworks>net5</TargetFrameworks>
-    <LangVersion>preview</LangVersion>
+    <TargetFramework>$(FrameworkBase)</TargetFramework>
   </PropertyGroup>
 
   <PropertyGroup Label="NuGet Package Information">
     <PackageId>OpenTabletDriver.Plugin</PackageId>
-    <Version>0.6.0.0</Version>
-    <Authors>InfinityGhost</Authors>
```
Comment (gonX) [inline OpenTabletDriver.Plugin/OpenTabletDriver.Plugin.csproj:11]: Removing these 2 lines seems unrelated to changing to .net6. What does this do?

Comment (X9VoiD) [reply OpenTabletDriver.Plugin/OpenTabletDriver.Plugin.csproj:11]: It's moved to `Directory.Build.props` so that property applies to all projects by default.

**Comment by RandomExplosion** [comment]
If you guys don't mind me asking, what's the status on this pr?
From what i can tell it seems to be ready (i don't know the code base of course tho so there may well be something missing)

**Comment by X9VoiD** [comment]
It's completely ready, it's just that @InfinityGhost wants to wait for osu! framework to also migrate to .NET 6 or atleast add support for it.

**Comment by zsinx6** [comment]
@X9VoiD can you give more details about what is the relation between the osu! framework and this?

**Comment by jamesbt365** [comment]
Our driver is integrated with osu lazer and framework, for compatibility reasons we will move to dotnet 6 when they move

**Comment by X9VoiD** [comment]
Added multi-targeting to `OpenTabletDriver` project so there is no need to worry about compatibility anymore.

**Review by InfinityGhost — APPROVED** [review]
